### ISSUE TITLE
SIM reconstruction problem

### ISSUE THREAD

**Issue by jumpxiu** [TristaZeng/ZS-DeconvNet#7]
I want to train a model from scratch using SIM nine graphs, and when I get stuck at step 3: "SIM reconstruction", which reconstruction algorithm exactly?

**Comment by TristaZeng** [comment]
You can use any open-source SIM reconstruction tools (e.g., openSIM) to reconstruct one SIM image from your nine graphs.

**Comment by jumpxiu** [comment]
> You can use any open-source SIM reconstruction tools (e.g., openSIM) to reconstruct one SIM image from your nine graphs.

I use the HIFI SIM algorithm for reconstruction. When I use the BioSR data after noise addition for SIM reconstruction, I get very bad results for levels 1-8! I am wondering what is the reason. Thank you for your answer!

**Comment by TristaZeng** [comment]
I'm not sure if by "use the BioSR data after noise addition for SIM reconstruction" you mean you added noise before reconstruction. Noise addition is conducted after SIM reconstruction in order to form paired train data for network training.

**Comment by jumpxiu** [comment]
> I'm not sure if by "use the BioSR data after noise addition for SIM reconstruction" you mean you added noise before reconstruction. Noise addition is conducted after SIM reconstruction in order to form paired train data for network training.

Sorry, I should have misunderstood the meaning of Supplementary Fig.16. Thanks again!

**Comment by jumpxiu** [comment]
> I'm not sure if by "use the BioSR data after noise addition for SIM reconstruction" you mean you added noise before reconstruction. Noise addition is conducted after SIM reconstruction in order to form paired train data for network training.

I am still puzzled that there is given in your website:
1.2 To generate training dataset for ZS-DeconvNet-SIM, you can:
-     Prepare a folder of raw data.
-     Open ./Python_MATLAB_Codes/data_augment_recorrupt_matlab/GenData4ZS-DeconvNet-SIM/Create_corrupt_img_2D.m or Create_corrupt_img_3D.m and set the parameters.
-     Perform SIM reconstructions on the corrupted dataset to generate the input and targets for ZS-DeconvNet-SIM.

I tried to use BioSR for the above steps, but the SIM reconstructions gave me absolutely unusable results!

**Comment by TristaZeng** [comment]
Sorry I gave a wrong reply a few days ago. Could you tell me the exact data you used, and whether you changed any parameters? Also let me see the reconstruction result.

**Comment by jumpxiu** [comment]
> Sorry I gave a wrong reply a few days ago. Could you tell me the exact data you used, and whether you changed any parameters? Also let me see the reconstruction result.

I used BioSR data and not modify any parameters in the MATLAB file(data recorrupt for SIM).

**Comment by TristaZeng** [comment]
Please give me an example of the input and your reconstruction result otherwise we cannot pinpoint the problem.

**Comment by jumpxiu** [comment]
> Please give me an example of the input and your reconstruction result otherwise we cannot pinpoint the problem.

The image I entered is ` 'BioSR\Microtubules\Cell_001\RawSIMData_gt.mrc'`. After using your reconstruction program, I get input1, 2, 3 and target1, 2, 3. After putting them through a HIFI SIM reconstruction I get a reconstruction that is all black. Or which dataset should I use to train SIM data? To get good results like yours.

**Comment by TristaZeng** [comment]
Input1, 2, 3 and target1, 2, 3 should be put separately through HIFI SIM reconstruction and got 6 results to form 3 data pairs. If the reconstructions are all black, the most likely reason is wrong reconstruction parameter estimation. We recommend using `BioSR\Microtubules\Cell_001\RawSIMData_gt.mrc` instead of the corrupted data to estimate the SIM parameters first.

**Comment by jumpxiu** [comment]
> Input1, 2, 3 and target1, 2, 3 should be put separately through HIFI SIM reconstruction and got 6 results to form 3 data pairs. If the reconstructions are all black, the most likely reason is wrong reconstruction parameter estimation. We recommend using `BioSR\Microtubules\Cell_001\RawSIMData_gt.mrc` instead of the corrupted data to estimate the SIM parameters first.

Thank you! I was also wondering if the deconv phase of training has any effect on the denoise phase of training?

**Comment by TristaZeng** [comment]
Intuitively the gradients backpropagate from the deconv phase to the denoise phase so the effect of the denoise phase is bound to be affected by deconv phase, and because of the joint optimizing goal they can help each other. But we have not designed specific experiments to compare the denoising results of ZS-DeconvNet and the denoising results without a deconv phase in the network.

**Comment by jumpxiu** [comment]
This looks like the result of denoising without the denoising stage in the network is not good like  the denoising results of ZS-DeconvNet. Thank you for your research.
